Server/JokesOnYou.Web.Api/Repositories/Interfaces/IJokesRepository.cs

```diff
@@ -7,5 +9,9 @@ namespace JokesOnYou.Web.Api.Repositories.Interfaces
 {
     public interface IJokesRepository
     {
+        public Task CreateJokeAsync(Joke joke);
+        public Task<IEnumerable<Joke>> GetAllJokesAsync();
+        public Task<IEnumerable<Joke>> GetJokesByPremiseAsync(string premise);
```

Comment:
- Yes I am typing them in, I know it does not matter. I like to explicitly type stuff so I don't make any mistakes of forgetting it. I like to get used to stuff, so wherever I can I'll type it.
- Yes its technically a "filter" Though I needed this to check if a joke already exists in the DB. so its not the whole filtering issue that can be done by someone else and then this could get integrated.